src/js/providers/tracks-mixin.ts

```diff
@@ -607,23 +608,48 @@ const Tracks: TracksMixin = {
             (lastCue._parsed || cuesMatch(previousCues[previousCues.length - 1], lastCue))) {
             return;
         }
+        const dataCueSets: Array<TrackCueParsed[]> = [];
+        const parsedDataCueSets: Array<TrackCueParsed[]> = [];
         let dataCueSetIndex = -1;
         let startTime = -1;
-        const dataCueSets: Array<TrackCueParsed[]> = Array.prototype.reduce.call(cues, (cueSets: Array<TrackCueParsed[]>, cue: TrackCueParsed) => {
-            if (!cue._parsed && ((cue as DataCue).data || cue.value)) {
+        let previousStart = -1;
+        for (let i = 0; i < cues.length; i++) {
+            const cue: TrackCueParsed = cues[i];
+            if (!cue._extended && !!((cue as DataCue).data || cue.value)) {
                 if (cue.startTime !== startTime || cue.endTime === null) {
+                    previousStart = startTime;
                     startTime = cue.startTime;
-                    cueSets[++dataCueSetIndex] = [];
+                    const previousSet = dataCueSets[dataCueSetIndex];
+                    dataCueSets[++dataCueSetIndex] = [];
+                    parsedDataCueSets[dataCueSetIndex] = [];
+                    // increase id3 cue duration to a minimum of 0.25s up to next id3 cue start to ensure
+                    // "cuechange" event is fired and it is added to activeCues when currentTime intersects cue range
```

Comment:
Is this comment more so for cueParsed logic below (Line 640)?

Reply:
`previousCue.endTime = startTime` changes the cue duration. but that block work starts on the next line.